src/services/data_model_engine.js

```diff
@@ -254,15 +255,10 @@ export default class DataModelEngine {
       throw new Error(`Could not fetch the bundle from the shelterer: ${err.message || err}`);
     }
 
-    await this.bundleRepository.setBundleRepository(bundleId, BundleStatusStates.downloaded, {nodeUrl});
-    await this.bundleRepository.storeBundleProofMetadata(
-      complementedMetadata.bundleId,
-      complementedMetadata.bundleProofBlock,
-      complementedMetadata.bundleUploadTimestamp,
-      complementedMetadata.bundleTransactionHash
-    );
+    await this.bundleRepository.setBundleRepository(bundleId, BundleStatuses.downloaded, {nodeUrl});
+    await this.bundleRepository.storeAdditionalMetadata(bundleId, downloadedMetadata);
 
-    return complementedMetadata;
+    return initialMetadata;
```

Comment:
Doesn't make much difference now, but it would be less surprising if we returned the complementedMetadata (meaning initial merged with downloaded).